doc/dev/doc/serialization/compact.md

```diff
@@ -0,0 +1,77 @@
+
+Currently, most primitive types (`bool`, `int`...) are supported, as well as enums, arrays of those types, and nested classes. More advanced classes (e.g. `List<T>`) are not supported yet. Intefaces are not supported.
+
+## Compact Serializer
+
+Another way to use compact serialization is to implement the `ICompactSerializer<T>` interface for a type, and register it in the configuration. A basic serializer could look like:
+
+```csharp
+public class EmployeeSerializer : ICompactSerializer<Employee>
+{
+    public string TypeName => "employee";
+
+    public Employee Read(ICompactReader reader)
+    {
+        return new Employee
+        {
+            Id = reader.ReadInt64("id");
+            Name = reader.ReadString("name");
+        }
+    }
+
+    public void Write(ICompactWriter writer, Employee employee)
+    {
+        writer.WriteInt64("id", employee.Id);
+        writer.WriteString("name", employee.Name);
+    }
+}
+```
+
+Then, that serializer must be registered in the configuration:
+
+```csharp
+options.Serialization.Compact.AddSerializer(new EmployeeSerializer());
+```
+
+> [!NOTE]
+> Only programmatic configuration is supported by the .NET client at the moment.
+
+## Schema Evolution
+
+Compact serialization permits schemas and classes to evolve by adding or removing fields, or by changing the types of fields. More than one version of a class may live in the same cluster and different clients or members might use different versions of the class.
+
+Hazelcast handles the versioning internally. So, you don’t have to change anything in the classes or serializers apart from the added, removed, or changed fields.
+
+Hazelcast achieves this by identifying each version of the class by a unique fingerprint. Any change in a class results in a different fingerprint. Hazelcast uses a 64-bit Rabin Fingerprint to assign identifiers to schemas, which has an extremely low collision rate.
+
+Different versions of the schema with different identifiers are replicated in the cluster and can be fetched by clients or members internally. That allows old readers to read fields of the classes they know when they try to read data serialized by a new writer. Similarly, new readers might read fields of the classes available in the data, when they try to read data serialized by an old writer.
+
+This means that for one *type name*, there can be several schemas.
+
+In addition, the `ICompactReader` interface exposes methods such as `FieldKind GetFieldKind(string name)` which returns the *kind* (i.e. the actual type) of the field. 
```

Comment:
I don't know 5.2 documentation is up to date (I am comparing yours with version [5.1](https://docs.hazelcast.com/hazelcast/5.1/serialization/compact-serialization)) but maybe we can add a schema evolution usage here. In version [5.2](https://docs.hazelcast.com/hazelcast/5.2-snapshot/serialization/compact-serialization), there is an example.

@zpqrtbnk